Python/suggestions.c

```diff
@@ -202,13 +202,22 @@ offer_suggestions_for_name_error(PyNameErrorObject *exc)
     PyTracebackObject *traceback = (PyTracebackObject *) exc->traceback; // borrowed reference
     // Abort if we don't have a variable name or we have an invalid one
     // or if we don't have a traceback to work with
-    if (name == NULL || traceback == NULL || !PyUnicode_CheckExact(name)) {
+    if (name == NULL || !PyUnicode_CheckExact(name) ||
```

Comment:
Instead of checking for None, we can check if the traceback object parent class is the traceback type. This way we also stop segfaults if someone placed any other object there

Reply:
Isn't tb_next is supposed to be [None or a Traceback](https://github.com/python/cpython/blob/main/Objects/exceptions.c#L236)? Regardless, I agree that it is more clear to just check for type.